pyfr/readers/cgns_wrapper.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+
+import os
+
+from ctypes import (RTLD_GLOBAL, c_size_t, c_int, c_void_p, c_char_p,
+                    byref, create_string_buffer)
+
+from pyfr.ctypesutil import load_library
+
+
+class CGNSWrapper:
+    def __init__(self):
+        self._hdf5_load()
+
+        # Load CGNS
+        self.lib = lib = load_library('cgns')
+
+        self.cgnsUnstructured = 3
+        self.Integer, self.RealSingle, self.RealDouble = 2, 3, 4
+
+        # error
+        lib.cg_get_error.argtypes = None
+        lib.cg_get_error.restype = c_char_p
+
+        # cg_open
+        lib.cg_open.argtypes = [c_char_p, c_int, c_void_p]
+        lib.cg_open.errcheck = self._errcheck
+
+        # cg_nbases
+        lib.cg_nbases.argtypes = [c_int, c_void_p]
+        lib.cg_nbases.errcheck = self._errcheck
+
+        # cg_nzones
+        lib.cg_nzones.argtypes = [c_int, c_int, c_void_p]
+        lib.cg_nzones.errcheck = self._errcheck
+
+        # cg_base_read
+        lib.cg_base_read.argtypes = [c_int, c_int, c_char_p, c_void_p, c_void_p]
+        lib.cg_base_read.errcheck = self._errcheck
+
+        # cg_coord_read
+        lib.cg_coord_read.argtypes = [c_int, c_int, c_int, c_char_p, c_int,
+                                      c_void_p, c_void_p, c_void_p]
+        lib.cg_coord_read.errcheck = self._errcheck
+
+        # cg_nbocos
+        lib.cg_nbocos.argtypes = [c_int, c_int, c_int, c_void_p]
+        lib.cg_nbocos.errcheck = self._errcheck
+
+        # cg_boco_info
+        lib.cg_boco_info.argtypes = [c_int, c_int, c_int, c_int, c_char_p,
```

Comment:
Use POINTER(c_int) where appropriate.  Avoids having to use byref.  See the METIS partitioner interface for further details.